---
layout: issue
title: "Action View Helper causes some strange issues with Exceptions"
id: ZF-6850
---

ZF-6850: Action View Helper causes some strange issues with Exceptions
----------------------------------------------------------------------

 Issue Type: Patch Created: 2009-05-28T11:38:37.000+0000 Last Updated: 2011-02-18T08:14:43.000+0000 Status: Postponed Fix version(s): 
 Reporter:  Corey Frang (gnarf)  Assignee:  Ralph Schindler (ralph)  Tags: - Zend\_View
 
 Related issues: 
 Attachments: - [View\_Helper\_Action.diff](/issues/secure/attachment/11961/View_Helper_Action.diff)
- [View\_Helper\_Action.diff](/issues/secure/attachment/11960/View_Helper_Action.diff)
 
### Description

The Zend\_View\_Helper\_Action has no exception Handling. If an exception is thrown in the action, and caught - the view renderer object never gets set back into place.

I wrote a little patch around the issue - it just catches the exception in the dispatch, and throws the same exception after setting the view stuff back to the original state.

 

 

### Comments

Posted by Corey Frang (gnarf) on 2009-05-28T12:02:15.000+0000

Updating the file - changed doc block to include @throws

 

 

Posted by Dolf Schimmel (Freeaqingme) (freak) on 2009-05-28T12:05:37.000+0000

You're aware that using the action viewhelper is considered bad practice? See also ZF-5840 or <http://tinyurl.com/actionstack>

 

 